### ISSUE TITLE
Form control value changes event getting invoked on clicking Add Button 

### ISSUE THREAD

**Issue by afsalkriti** [ngx-formly/ngx-formly#3130]
**Description**
Whenever the add button is clicked on an repeat component  , 
The the value change event of the fields gets invoked ( even without changing ) and new values gets assigned.

**Expected Behaviour**
Need to keep the values of the model same , and need to update value only if model changed.

**Minimal Reproduction**

[https://stackblitz.com/edit/angular-zdfsty?file=src%2Fapp%2Fapp.component.ts](https://stackblitz.com/edit/angular-zdfsty?file=src%2Fapp%2Fapp.component.ts)

**Your Environment**
- Production Version: 12
- Test Angular version: 7
- Formly version: 5.10.29

**Comment by aitboudad** [comment]
for a workaround use `distinctUntilChanged`:
```ts
field.formControl.valueChanges.pipe(distinctUntilChanged())
```

**Comment by afsalkriti** [comment]
The Workaround is also not working fine , it still changes the value .

**Comment by aitboudad** [comment]
seems working on my side, could please let me know how I can reproduce it?
I've updated your example to trigger the event during the initial subscribe using `startWith` operator https://stackblitz.com/edit/angular-zdfsty-caruan?file=src/app/app.component.ts

**Comment by aitboudad** [comment]
after rechecking your example, I see the issue now, working on a fix!

**Comment by aitboudad** [comment]
This issue has been fixed and released as part of [v5.10.30](https://github.com/ngx-formly/ngx-formly/releases/tag/v5.10.30) release.

Please let us know, in case you are still encountering a similar issue/problem.
Thank you!
